test/contrib/test_pyopenssl_dependencies.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+import unittest
+
+from nose.plugins.skip import SkipTest
+
+try:
+    from urllib3.contrib.pyopenssl import inject_into_urllib3
+except ImportError as e:
+    raise SkipTest('Could not import PyOpenSSL: %r' % e)
+
+from mock import patch
+
+class TestPyOpenSSLInjection(unittest.TestCase):
+    """
+    Tests for error handling in pyopenssl's 'inject_into urllib3'
+    """
+    def test_inject_validate_fail(self):
+        """
+        Injection should not be supported if we are missing required dependencies.
+        """
+        with patch("cryptography.x509.extensions.Extensions") as mock:
+            del mock.get_extension_for_class
+            self.assertRaises(ImportError, inject_into_urllib3)
```

Comment:
Hm...  Is `extract_from_urllib3` supposed to always be safe to run even if `inject_into_urllib3` has not been called?  I wasn't assuming that was part of the API's contract -- I assumed it was more likely that `inject_into_urllib3` would correctly handle repeated calls because developers would be more likely to call it by mistake, hence putting this test in the other file.

I can add a try/except/finally block that cleans up only if no assertion is raised.  That would be less than elegant, but it would arguably be more correct...